### ISSUE TITLE
Faster R-CNN training with negative images(non-object image)

### ISSUE THREAD

**Issue by naeunhub** [pytorch/vision#5098]
### 📚 The doc issue

I'm using torchvision.models.detection.fasterrcnn to train my custom dataset.
For annotations that do not have any bounding box, I use target [0, 0, 0, 0], label 0.
But it return ValueError(All bounding boxes should have positive height and width).

And when I use target [0, 0, 0.5, 0.5], label 0.
But it returns nan Loss..

Is there any solution for this problem?

**Comment by jveitchmichaelis** [comment]
I've also come up against this when performing fairly heavy augmentation + cropping on images that have sparse masks. It's not uncommon to get a transformed image with no objects in it. I'm not sure how Detectron2 handles this.

One optionn in frameworks like Lightning is to ignore the problem by checking to see if the batch has any empty images/masks and just returning `None`, but you'll get a warning.

For an empty image, our goal is to penalise the model for producing any bounding box. One hacky approach would be to append a single pixel target at a random location to prevent the model from trying to learn it (if you always choose e.g. [0,0] the model will probably try and predict a box there every time).

```python
if no_objects:
    target = {}
    b, c, h, w = images.shape
    n_class = 2
    x = random.randint(0, w-1)
    y = random.randint(0, h-1)
    
    target['masks'] = torch.zeros(size=(1,h,w)).bool()
    target['boxes'] = torch.tensor([[x,y,x+1,y+1]]).float()
    target['labels'] = torch.tensor([random.randint(0, n_class)]).long()

    targets.append(target)
```

If the model returns predictions, then the loss should be high as desired.
